docs/bbe-guidelines/bbe-guidelines.md

```diff
@@ -0,0 +1,122 @@
+
+        <img src="images/keyword-in-bio.png" width="400" height="80" alt="keywords in bio" />
+
+    > **Note:** However, the context should be taken into account when formatting words/phrases with backquotes. For example, although the word "record" is a keyword of the language, it need not be highlighted if you are referring to the record construct of the language as shown below.
+
+    ````
+    // Defines an open record type named Student. The { and } delimiters indicate that in addition to the defined fields,
+    // this record type allows additional fields with pure-typed (i.e., anydata|error) values.
+    // The descriptor record { } is equivalent to record {| (anydata|error)...; |}.
+    type Student record {
+        string name;
+        int age;
+    Grades grades;
+    };
+    ```
+
+7. Remove unused imports in `.bal` files.
+
+8. If a new example is added or an existing one is deleted/renamed, update the [index.json](https://github.com/ballerina-platform/ballerina-lang/blob/master/examples/index.json) file as well.
+
+## Practices related to the code comments
+
+1. Add comments to the code blocks as much as possible by starting them with “//”. They will be moved to the right hand side boxes in the [Ballerina By Example](https://ballerina.io/learn/by-example/) as a mechanism to describe the code.
+
+    Example;
+
+    - In the code:
+
+        <img src="images/code-comments.png" width="600" height="80" alt="adding code comments" />
+
+    - In the [Ballerina website](https://ballerina.io/):
+
+        ![comment boxes](images/comment-boxes.png "comment-boxes")
+
+2. Since comments are used in the right hand side boxes as a mechanism to describe the code, they should be valid sentences (i.e., start with an upper case letter and end with a full stop etc.).
+
+3. After a comment is strated, it will continue being applied to the subsequent lines in the file until another comment or an empty line is found. Therefore, use comments/new lines appropriately to ensure that they apply only to the relevant lines.
+
+4. There is no restriction on the max number of characters that should be there in comment lines as they get wrapped automatically in the right hand side in the [Ballerina By Example](https://ballerina.io/learn/by-example/). However, since users can refer the same in the code in GitHub, it is better if we can have the same character limit as of a code line (i.e., 80) to increase readability.
+
+    For example, it will not be readable if comments are significantly longer than the code line as shown below. 
+
+    ![long comments](images/long-comments.png "long-comments")
+
+## Practices related to the folder structure
+
+1. Each BBE should be in a separate directory in the [ballerina-lang Git Hub repository](https://github.com/ballerina-platform/ballerina-lang/tree/master/examples) named by the example (e.g., https://github.com/ballerina-platform/ballerina-lang/tree/master/examples/artemis-transaction-producer).
```

Comment:
```suggestion
1. Each BBE should be in a separate directory in the [ballerina-lang GitHub repository](https://github.com/ballerina-platform/ballerina-lang/tree/master/examples) named by the example (e.g., https://github.com/ballerina-platform/ballerina-lang/tree/master/examples/artemis-transaction-producer).
```

Reply:
Also shall we use a more common example such as hello-world-service as the example link?

Reply:
Fixed "GitHub" and added [1] as the link.

[1] https://github.com/ballerina-platform/ballerina-lang/tree/master/examples/hello-world